src/reducers.js

```diff
@@ -0,0 +1,43 @@
+import { Record } from 'immutable'
+import _ from 'lodash'
+import { workspaceAccessLevels } from 'src/libs/utils'
+
+
+export const initialState = new (Record({
+  workspaces: new (Record({
+    filter: '',
+    listView: false,
+    itemsPerPage: 6,
+    pageNumber: 1,
+    workspaces: null,
+    failure: undefined,
+  }))()
+}))()
+
+export const reducers = {
```

Comment:
There are many options for organizing the reducer function. This has everything in one file, but you could also split it up by store sub-key, page, category, or anything else.